jsonrpc-specs/src/methods/state.json

```diff
@@ -0,0 +1,490 @@
+[
+  {
+    "name": "portal_stateRoutingTableInfo",
+    "summary": "Returns meta information about state network routing table.",
+    "params": [],
+    "result": {
+      "name": "stateTableInfoResult",
+      "description": "State network routing table information",
+      "required": true,
+      "schema": {
+        "title": "overlayTableInfoResults",
+        "description": "Routing table details",
+        "type": "object",
+        "required": [
+          "localNodeId",
+          "buckets"
+        ],
+        "properties": {
+          "localNodeId": {
+            "title": "nodeId",
+            "description": "The key identifying the local peer that owns the routing table.",
+            "$ref": "#/components/schemas/bytes32"
+          },
+          "buckets": {
+            "title": "kBucketsTable",
+            "description": "Represents a Kademlia routing table.",
+            "$ref": "#/components/schemas/kBuckets"
+          }
+        }
+      }
+    }
+  },
+  {
+    "name": "portal_stateAddEnr",
+    "summary": "Write an ethereum node record to the routing table.",
```

Comment:
Is there any way that we can reduce the duplication that is currently present between things like `portal_historyAddENR` and `portal_stateAddENR`?

Reply:
I managed to add some additional schemas for the common method results, which removes some of the duplications. 

Params is not a JSON-Schema, but an array of [Content Descriptor Objects](https://spec.open-rpc.org/#content-descriptor-object). It seems we can reference only the schemas and not the whole array of params which prevents us from removing the params duplications. Maybe there is another way to do this, but not sure if it is worth the effort at this stage...